freecad/Curves/Sketch_On_Surface.py

```diff
@@ -445,7 +445,13 @@ def build_sketch(sk, fa):
     elif isinstance(fa.Surface, Part.OffsetSurface) and isinstance(fa.Surface.BasisSurface, Part.SurfaceOfExtrusion) and isinstance(fa.Surface.BasisSurface.BasisCurve, Part.BSplineCurve):
         u0 = 0
         u1 = fa.Surface.BasisSurface.BasisCurve.length()
-
+    elif isinstance(fa.Surface, Part.SurfaceOfRevolution) and isinstance(fa.Surface.BasisCurve, Part.BSplineCurve):
```

Comment:
Why restrict to BSpline Basis curve ?
**length** method is available from all geometry curves.

Reply:
It is just an extended copy&paste of the block above (`SurfaceOfExtrusion`). It worked for my use-case and would (probably) work for the linked issue, but I don't say I really fully understand whole picture behind all the geometry parts.

Reply:
OK.
The default construction of the mapped_sketch has been bugged for years.
I will have a look to see if there is a better general solution.